### DISCUSSION TITLE
Welcome to chess.com Discussions!

### DISCUSSION THREAD

**Discussion by scaljeri** [scaljeri/chess.com#15] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MihailOsincev** [comment]
Hello, I am a novice developer in C Sharp, please tell us how you read the positions of the chess pieces ?

**Reply by scaljeri** [reply]
Hi

I actually don't read positions, I parse the game history, shown to the right of the chess board

**Reply by MihailOsincev** [reply]
I looked at how it works,with each move a new div container is generated,and you get to catch this event and process it?

**Reply by MihailOsincev** [reply]
And if you don't mind you can specify in which file you have this handler

**Reply by scaljeri** [reply]
https://github.com/scaljeri/chess.com/blob/master/packages/bot/src/frontend/utils/game-history.ts#L59

**Comment by timlecou** [comment]
Hi, you have pushed your chess.com credentials in the repo, make sure to remove it before someone uses it wrong

**Comment by scaljeri** [comment]
Thnx, I've removed them!

**Comment by Redflame1022** [comment]
Not sure if I will get a response as im a bit late, however I have an issue. Whenever I try to build the script I get 

```
lerna info run Ran npm script 'build' in '@scaljeri/chess-shared' in 1.1s:
yarn run v1.22.18
$ rm -rf ./dist && tsc
Done in 0.93s.
lerna ERR! yarn run build exited 1 in '@scaljeri/chess-com-bot'
lerna ERR! yarn run build stdout:
yarn run v1.22.18
$ rm -rf ./dist && yarn build:frontend && yarn build:backend && yarn copy-api-worker
$ di -d -c tsc -b ./src -i shared frontend/main.ts -- -p tsconfig-frontend.json && yarn build:browserify
Glob file search: src\**\*.ts
Adding src/shared/environments/komodo.ts
Adding src/shared/game-play.ts
Adding src/shared/turn-observer.ts
Task skipped, content unchanged (src\frontend\main-di.ts)
Running: tsc -p tsconfig-frontend.json
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run build stderr:
node:events:505
      throw er; // Unhandled 'error' event
      ^

Error: spawn tsc ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn tsc',
  path: 'tsc',
  spawnargs: [ '-p', 'tsconfig-frontend.json' ]
}

Node.js v17.7.2
error Command failed with exit code 1.
error Command failed with exit code 1.
```

**Reply by scaljeri** [reply]
Is this on Windows right? If I remember correctly this is the error you get on windows. Some of the commands in the build tooling are not cross platform. It will work on a Mac and I think Linux will work too. Sorry for that.

**Comment by Theodoregore** [comment]
What sort of programming hete

**Reply by Theodoregore** [reply]
What sort of programming here
